build/Targets/Settings.targets

```diff
@@ -128,17 +128,31 @@
       </PropertyGroup>
     </Otherwise>
   </Choose>
+  
+  <!-- This is a really hacky way to detect whether we are on a legacy or a willow based VS install.
+       Basically, we check for a registry key that will only exist in legacy VS installs, and assume
+       we are a willow based installation if our VSVersion is 15.0 and the registry key doesn't exist. -->
```

Comment:
I thought about that, but we need to work if the user explicitly calls MSBuild as well (for example, outside of the Developer Command Prompt).